apps/etterlatte-vedtaksvurdering/src/main/kotlin/no/nav/etterlatte/vedtaksvurdering/VedtakBehandlingService.kt

```diff
@@ -492,16 +496,34 @@ class VedtakBehandlingService(
                         utbetalingsperioder =
                             opprettUtbetalingsperioder(
                                 vedtakType = vedtakType,
-                                virkningstidspunkt = virkningstidspunkt,
                                 beregningOgAvkorting = beregningOgAvkorting,
                                 behandling.sakType,
+                                opphoerFraOgMed = opphoerFraOgMed,
                             ),
                         revurderingInfo = behandling.revurderingInfo,
+                        opphoerFraOgMed = opphoerFraOgMed,
                     ),
             )
         return repository.oppdaterVedtak(oppdatertVedtak)
     }
 
+    private fun utledOpphoerFraOgMed(
+        vedtakType: VedtakType,
+        virkningstidspunkt: YearMonth,
+        behandling: DetaljertBehandling,
+    ) = when (vedtakType) {
+        VedtakType.OPPHOER -> {
+            virkningstidspunkt
+        }
+        else -> {
+            if (virkningstidspunkt == behandling.opphoerFraOgMed) {
```

Comment:
Burde det være noe mer sjekking her tro? Hva om feks virkningstidspunktet er etter opphoerFraOgMed? Eller burde det feile lenge før i flyten kanskje?

Reply:
Ja tenkte at det burde være en valideringsfeil når man setter virk. Men skader ikke å ha en ekstra sjekk her for sikkerhets skyld. Legger det til.